spec/helpers/script_helper_spec.rb

```diff
@@ -33,28 +33,27 @@
         javascript_packs_tag_once('application', prepend: true)
       end
 
-      it 'prints all unique packs in order' do
+      it 'prints all unique packs in order, locale scripts first' do
         output = render_javascript_pack_once_tags
-
-        application_pack_selector = 'script[src^="/packs-test/js/application-"]'
-        clipboard_pack_selector = 'script[src^="/packs-test/js/clipboard-"]'
-        document_capture_pack_selector = 'script[src^="/packs-test/js/document-capture-"]'
-
-        expect(output).to have_css(
-          application_pack_selector,
-          count: 1,
-          visible: false,
-        )
-        expect(output).to have_css(
-          "#{application_pack_selector} ~ #{clipboard_pack_selector}",
-          count: 1,
-          visible: false,
-        )
-        expect(output).to have_css(
-          "#{clipboard_pack_selector} ~ #{document_capture_pack_selector}",
-          count: 1,
-          visible: false,
-        )
+        public_output_path = current_webpacker_instance.config.public_output_path
+        public_path = current_webpacker_instance.config.public_path
+        output_path = public_output_path.relative_path_from(public_path)
+
+        selectors = [
+          "script[src^='/#{output_path}/js/application-'][src$='.chunk.en.js']",
+          "script[src^='/#{output_path}/js/document-capture-'][src$='.chunk.en.js']",
+          "script[src^='/#{output_path}/js/runtime~application-']",
+          "script[src^='/#{output_path}/js/application-'][src$='.chunk.js']",
+          "script[src^='/#{output_path}/js/clipboard-'][src$='.chunk.js']",
+          "script[src^='/#{output_path}/js/document-capture-'][src$='.chunk.js']",
+        ]
+
+        selectors.each_with_index do |selector, i|
+          next_selector = selectors[i + 1]
+          test_selector = selector
+          test_selector += " ~ #{next_selector}" if next_selector
+          expect(output).to have_css(test_selector, count: 1, visible: false)
+        end
```

Comment:
Can we rely on script tags coming back in order from Nokogiri? Would a simpler test something like:

```ruby
script_srcs = Nokogiri::HTML(output).css('script').map { |s| s[:src] }

expect(script_srcs).to all(start_with("/#{output_path}/js"))

# relies on order?
script_srcs.zip([
  %r{/application-[a-f0-9]+.chunk.en.js$}
  %r{/document-capture-[a-f0-9]+.chunk.en.js$}
  %r{/runtime-application-[a-f0-9]+.js$}
]).each do |src, regex|
  expect(src).to match(regex)
end
```

Reply:
It is in order, yes, but the problem is that this isn't the _exclusive_ set of scripts that we'd expect. The way that chunking works is that Webpack will try to find an optimal way to group together common dependencies amongst the various packs that we have.

Example:

```html
<script src="/packs/js/application-9eac385e5a76ae977d07.chunk.en.js"></script>
<script src="/packs/js/form-validation-58e40285226b0488a3a2.chunk.en.js"></script>
<script src="/packs/js/runtime~application-aa001ceb574a6f67800d.js"></script>
<script src="/packs/js/vendors~accept-terms-button~application~clipboard~doc-capture-polling~document-capture~document-capt~f18d40d9-5a34116e2b85240f240b.chunk.js"></script>
<script src="/packs/js/vendors~application~doc-capture-polling~document-capture~document-capture-welcome~email-validation~f~69019cad-46d46e446dc35fade1c8.chunk.js"></script>
<script src="/packs/js/vendors~application~doc-capture-polling~document-capture~form-steps-wait~form-validation~intl-tel-in~4a3be832-62d9e46e1731a305fa0a.chunk.js"></script>
<script src="/packs/js/vendors~application~doc-capture-polling~document-capture~intl-tel-input~personal-key-page-controller~9898f78d-0e2f7e278fba7ff53b00.chunk.js"></script>
<script src="/packs/js/vendors~application~document-capture~webauthn-authenticate~webauthn-setup~webauthn-unhide-signin~web~0cc5b2b4-e693efa5350cd9d36797.chunk.js"></script>
<script src="/packs/js/vendors~application~document-capture~form-steps-wait~pw-strength~spinner-button-c20d1491dde5ef69515e.chunk.js"></script>
<script src="/packs/js/vendors~application~document-capture~intl-tel-input~personal-key-page-controller-bcf40ecce554912f23fd.chunk.js"></script>
<script src="/packs/js/vendors~application~document-capture-3d40bd3c971bd2b19dbb.chunk.js"></script>
<script src="/packs/js/vendors~application-ea1be68c85cbcd82f5fd.chunk.js"></script>
<script src="/packs/js/application-9eac385e5a76ae977d07.chunk.js"></script>
<script src="/packs/js/runtime~form-validation-f925453dd9af8963acd8.js"></script>
<script src="/packs/js/vendors~doc-capture-polling~document-capture~document-capture-welcome~form-steps-wait~form-validatio~548e369d-2ea7232ea2e44e0afffa.chunk.js"></script>
<script src="/packs/js/vendors~doc-capture-polling~document-capture~form-steps-wait~form-validation~intl-tel-input-995324e291998c6423c1.chunk.js"></script>
<script src="/packs/js/form-validation-58e40285226b0488a3a2.chunk.js"></script>
<script src="/packs/js/runtime~session-expire-session-a10c8f176b1a5b9a5c0e.js"></script>
<script src="/packs/js/session-expire-session-b5174fb5c378edecf23b.chunk.js"></script>
```

We can't predict how that chunking will turn out, but we can predict that the general order should be (1) locale data script, then (2) the runtime for the script, then (3) common "vendors" shared, then (4) the script itself.

Above:

```html
<script src="/packs/js/application-9eac385e5a76ae977d07.chunk.en.js"></script>
<!-- ... -->
<script src="/packs/js/runtime~application-aa001ceb574a6f67800d.js"></script>
<!-- ... -->
<script src="/packs/js/application-9eac385e5a76ae977d07.chunk.js"></script>
```

Reply:
dang, this is good to know, thanks for illustrating